src/govuk/components/header/header.mjs

```diff
@@ -8,34 +8,67 @@ function Header ($module) {
   this.$menu = this.$menuButton && $module.querySelector(
     '#' + this.$menuButton.getAttribute('aria-controls')
   )
+
+  // Save the opened/closed state for the nav in memory so that we can
+  // accurately maintain state when the screen is changed from small to
+  // big and back to small
+  this.menuIsOpen = false
+
+  // A global const for storing a matchMedia instance which we'll use to
+  // detect when a screen size change happens. We set this later during the
+  // init function and rely on it being null if the feature isn't available
+  // to initially apply hidden attributes
+  this.mql = null
 }
 
 /**
  * Initialise header
  *
  * Check for the presence of the header, menu and menu button – if any are
  * missing then there's nothing to do so return early.
+ * Feature sniff for and apply a matchMedia for desktop which will
+ * trigger a state sync if the browser viewport moves between states. If
+ * matchMedia isn't available, hide the menu button and present the "no js"
+ * version of the menu to the user.
  */
 Header.prototype.init = function () {
   if (!this.$module || !this.$menuButton || !this.$menu) {
     return
   }
 
-  this.syncState(this.$menu.classList.contains('govuk-header__navigation-list--open'))
-  this.$menuButton.addEventListener('click', this.handleMenuButtonClick.bind(this))
+  if ('matchMedia' in window) {
+    this.mql = window.matchMedia('(max-width: 48.0625em)')
```

Comment:
If you set the viewport to exactly 769px you get the menu button showing with the desktop layout:

<img width="783" alt="Screenshot 2022-08-01 at 14 57 33" src="https://user-images.githubusercontent.com/121939/182164327-7303fd60-d533-4f46-8e91-9fe777e92761.png">

Reply:
This seemed to be solved by changing the `em` value to `px`.

Reply:
After a chat offline with @36degrees we've managed to get this working whilst maintaining `em`'s by switching it to `min-width`.